cmd/wc/wc.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+func check(e error) {
+	if e != nil {
+		log.Fatal(e)
+	}
+}
+
+func main() {
+	var (
+		lines      bool
+		words      bool
+		characters bool
+	)
+	flag.BoolVar(&lines, "l", false, "lines")
+	flag.BoolVar(&words, "w", false, "words")
+	flag.BoolVar(&characters, "c", false, "characters")
+
+	flag.Parse()
+	path := flag.Args()[0]
+
+	absPath, err := filepath.Abs(path)
+	check(err)
+
+	if lines {
+		readFileLines, err := os.Open(absPath)
+		check(err)
+		lineScanner := bufio.NewScanner(readFileLines)
+		lineScanner.Split(bufio.ScanLines)
+		linesC := 0
+		for lineScanner.Scan() {
+			linesC++
+		}
+		fmt.Println("lines count: ", linesC)
+	}
+	if words {
+		readFileWords, err := os.Open(absPath)
+		check(err)
+		wordScanner := bufio.NewScanner(readFileWords)
```

Comment:
you can do this once for lines and words, this means if -l and -w are used you'll open the same file twice and don't close any of them